post_office/sanitizer.py

```diff
@@ -0,0 +1,144 @@
+            'table': ['class', 'id', 'style', 'align', 'bgcolor', 'border', 'cellpadding', 'cellspacing', 'dir', 'frame', 'rules', 'width'],
+            'tbody': ['class', 'id', 'style'],
+            'tfoot': ['class', 'id', 'style'],
+            'td': ['class', 'id', 'style', 'abbr', 'align', 'bgcolor', 'colspan', 'dir', 'height', 'lang', 'rowspan', 'scope', 'style', 'valign', 'width'],
+            'th': ['class', 'id', 'style', 'abbr', 'align', 'background', 'bgcolor', 'colspan', 'dir', 'height', 'lang', 'scope', 'style', 'valign', 'width'],
+            'thead': ['class', 'id', 'style'],
+            'tr': ['class', 'id', 'style', 'align', 'bgcolor', 'dir', 'style', 'valign'],
+            'u': ['class', 'id', 'style'],
+            'ul': ['class', 'id', 'style', 'dir', 'type'],
+        },
+        strip=True,
+        strip_comments=True,
+        styles=[
+            'border', 'border-top', 'border-right', 'border-bottom', 'border-left',
+            'border-radius',
+            'box-shadow',
+            'height',
+            'margin', 'margin-top', 'margin-right', 'margin-bottom', 'margin-left',
+            'padding', 'padding-top', 'padding-right', 'padding-bottom', 'padding-left',
+            'width',
+            'max-width',
+            'min-width',
+            'border-collapse',
+            'border-spacing',
+            'caption-side',
+            'empty-cells',
+            'table-layout',
+            'direction',
+            'font',
+            'font-family',
+            'font-style',
+            'font-variant',
+            'font-size',
+            'font-weight',
+            'letter-spacing',
+            'line-height',
+            'text-align',
+            'text-decoration',
+            'text-indent',
+            'text-overflow',
+            'text-shadow',
+            'text-transform',
+            'white-space',
+            'word-spacing',
+            'word-wrap',
+            'vertical-align',
+            'color',
+            'background',
+            'background-color',
+            'background-image',
+            'background-position',
+            'background-repeat',
+            'bottom',
+            'clear',
+            'cursor',
+            'display',
+            'float',
+            'left',
+            'opacity',
+            'outline',
+            'overflow',
+            'position',
+            'resize',
+            'right',
+            'top',
+            'visibility',
+            'z-index',
+            'list-style-position',
+            'list-style-tyle',
+        ],
+    ))
+except ImportError:
+    # if bleach is not installed, handle rendered HTML as plain text
+    clean_html = lambda body: mark_safe(format_lazy('<p><em>({heading})</em></p><div>{body}</div>',
+                                                    heading=gettext_lazy("stripping all HTML tags"),
+                                                    body=strip_tags(body)))
```

Comment:
One last comment. If bleach is not installed, we shouldn't preview any HTML email, even after `strip_tags`. We should just make a note that if previes is desired, install bleach. Django's docs [cautions against this](https://docs.djangoproject.com/en/3.0/ref/utils/#django.utils.html.strip_tags).

![Django Utils | Django documentation | Django 2020-06-16 07-26-46](https://user-images.githubusercontent.com/188658/84718592-2431e080-afa3-11ea-8a70-1cc72f73e282.jpg)

Reply:
Fixed by first stripping tags, then escaping. The other way would create an unreadable mess.